### PR TITLE
Some promise cleanup

### PR DESCRIPTION
**Pull request by aaomidi**
This PR fixes #

### CODE REVIEW

File: src/sql/platform/query/common/queryRunner.ts [hunk 1]
```diff
@@ -347,11 +347,10 @@ export default class QueryRunner extends Disposable {
 			let hasShowPlan = !!result.resultSetSummary.columnInfo.find(e => e.columnName === 'Microsoft SQL Server 2005 XML Showplan');
 			if (hasShowPlan) {
 				this._isQueryPlan = true;
-				this.getQueryRows(0, 1, result.resultSetSummary.batchId, result.resultSetSummary.id).then(e => {
-					if (e.resultSubset.rows) {
-						this._planXml.resolve(e.resultSubset.rows[0][0].displayValue);
-					}
-				});
+				const e = await this.getQueryRows(0, 1, result.resultSetSummary.batchId, result.resultSetSummary.id);
```
Comment (anthonydresser) [inline src/sql/platform/query/common/queryRunner.ts]: Don't think we want to await here, this code can safely be run out of context of the rest of this code.

File: src/sql/platform/query/common/queryRunner.ts [hunk 2]
```diff
@@ -373,20 +372,19 @@ export default class QueryRunner extends Disposable {
...
-						this._planXml.resolve(e.resultSubset.rows[0][0].displayValue);
-						// fire query plan available event if execution is completed
-						if (result.resultSetSummary.complete) {
-							this._onQueryPlanAvailable.fire({
-								providerId: mssqlProviderName,
-								fileUri: result.ownerUri,
-								planXml: planXmlString
-							});
-						}
+				const e = await this.getQueryRows(0, 1, result.resultSetSummary.batchId, result.resultSetSummary.id);
```
Comment (anthonydresser) [inline src/sql/platform/query/common/queryRunner.ts]: Same as the comment above.

File: src/sql/platform/query/common/queryRunner.ts (hunk 1 above)
Comment (aaomidi) [reply src/sql/platform/query/common/queryRunner.ts]: Will sync offline

File: src/sql/platform/query/common/queryRunner.ts [hunk 3]
```diff
@@ -347,11 +347,12 @@ export default class QueryRunner extends Disposable {
 			let hasShowPlan = !!result.resultSetSummary.columnInfo.find(e => e.columnName === 'Microsoft SQL Server 2005 XML Showplan');
 			if (hasShowPlan) {
 				this._isQueryPlan = true;
+
 				this.getQueryRows(0, 1, result.resultSetSummary.batchId, result.resultSetSummary.id).then(e => {
 					if (e.resultSubset.rows) {
 						this._planXml.resolve(e.resultSubset.rows[0][0].displayValue);
 					}
-				});
+				}).catch(console.error);
```
Comment (anthonydresser) [inline src/sql/platform/query/common/queryRunner.ts]: Use the logservice over just calling console

File: src/sql/platform/query/common/queryRunner.ts [hunk 4]
```diff
@@ -386,7 +388,7 @@ export default class QueryRunner extends Disposable {
 							});
 						}
 					}
-				});
+				}).catch(console.error);
```
Comment (anthonydresser) [inline src/sql/platform/query/common/queryRunner.ts]: logservice

File: src/sql/platform/query/common/gridDataProvider.ts [hunk 5]
```diff
@@ -23,7 +23,7 @@ export interface IGridDataProvider {
 	 * @param resultId The result id of the result to copy from
 	 * @param includeHeaders [Optional]: Should column headers be included in the copy selection
 	 */
-	copyResults(selection: Slick.Range[], includeHeaders?: boolean): void;
+	copyResults(selection: Slick.Range[], includeHeaders?: boolean): Promise<void>;
```
Comment (abist) [inline src/sql/platform/query/common/gridDataProvider.ts]: want to add `async` here?

Comment (aaomidi) [reply src/sql/platform/query/common/gridDataProvider.ts]: Don't think you can for method definitions.

Comment (anthonydresser) [reply src/sql/platform/query/common/gridDataProvider.ts]: async is more of a functional keyword than a typing keyword, so having something like 
`async function: Promise<void>` would be unnecessary.

File: src/sql/platform/query/common/queryRunner.ts [hunk 6]
```diff
@@ -347,11 +349,12 @@ export default class QueryRunner extends Disposable {
 			let hasShowPlan = !!result.resultSetSummary.columnInfo.find(e => e.columnName === 'Microsoft SQL Server 2005 XML Showplan');
 			if (hasShowPlan) {
 				this._isQueryPlan = true;
+
 				this.getQueryRows(0, 1, result.resultSetSummary.batchId, result.resultSetSummary.id).then(e => {
 					if (e.resultSubset.rows) {
 						this._planXml.resolve(e.resultSubset.rows[0][0].displayValue);
 					}
-				});
+				}).catch(this._logService.error);
```
Comment (abist) [inline src/sql/platform/query/common/queryRunner.ts]: should it be 
```.catch((e) => this._logService.error(e))``` instead ?

File: src/sql/platform/query/common/gridDataProvider.ts (hunk 5 above)
Comment (abist) [reply src/sql/platform/query/common/gridDataProvider.ts]: ah i didn't see it was a method def

File: src/sql/platform/query/common/queryRunner.ts [hunk 7]
```diff
@@ -386,7 +390,7 @@ export default class QueryRunner extends Disposable {
 							});
 						}
 					}
-				});
+				}).catch(this._logService.error);
```
Comment (abist) [inline src/sql/platform/query/common/queryRunner.ts]: same here

File: src/sql/platform/query/common/queryRunner.ts (hunk 6 above)
Comment (aaomidi) [reply src/sql/platform/query/common/queryRunner.ts]: e gets implicitly passed in i think

Comment (aaomidi) [reply src/sql/platform/query/common/queryRunner.ts]:
```
(async() => {
    throw new Error('boop');
})().catch(console.error);
```

yep

Comment (anthonydresser) [reply src/sql/platform/query/common/queryRunner.ts]: In either case you are passing through a function so it doesn't really matter, but `.catch((e) => this._logService.error(e))` is a little more explicit and easier to understand.

**Comment by anthonydresser** [comment]
Ensure you test these paths since some of these changes are functional changes

**Pull request merged**